modules/packages/LinearAlgebra.chpl

```diff
@@ -17,32 +17,139 @@
+default, the implementation with dependencies will be selected. Users can
+explicitly opt out of using the :mod:`BLAS` and :mod:`LAPACK` dependendent
+implentations by setting the ``blasImpl`` and ``lapackImpl`` flags to ``none``.
+
+**Building programs with no dependencies**
+
+.. code-block:: chpl
+
+  // example1.chpl
+  var A = Matrix([0.0, 1.0, 1.0],
+                 [1.0, 0.0, 1.0],
+                 [1.0, 1.0, 0.0]);
+  var I = eye(3,3);
+  var B = A + I;
+
+The program above has no dependencies and can therefore be compiled without
+the ``BLAS`` or ``LAPACK`` headers and libraries available:
+
+.. code-block:: bash
+
+  chpl example1.chpl
+
+If this program had used a procedure with a dependency such as
+:proc:`cholesky` (depends on ``LAPACK``), compilation without ``LAPACK``
+headers and libraries available would result in a compilation error.
+
+**Building programs with dependencies**
+
+.. code-block:: chpl
+
+  // example2.chpl
+  var A = Matrix([2, 1],
+                 [1, 2],
+                 eltType=real);
+  var (eigenvalues, eigenvectors) = eigvals(A, right=true);
+
+The program above uses :proc:`eigvals`, which depends on :mod:`LAPACK`.
+Compilation without ``LAPACK`` headers and libraries available would result in
+a compilation error.
+
+Following the instructions from the :mod:`LAPACK` module
+documentation, the above program could be compiled if ``LAPACK`` is available
+on the system and specified with the following compilation flags:
 
-To explicitly opt out of using the :mod:`BLAS` and :mod:`LAPACK` procedures, compile
-your Chapel program with the flags:
+.. code-block:: bash
+
+  # Building with LAPACK dependency
+  chpl -I$PATH_TO_LAPACKE_INCLUDE_DIR \
+       -L$PATH_TO_LIBGFORTRAN -lgfortran \
+       -L$PATH_TO_LAPACK_BINARIES -llapacke -llapack -lrefblas \
+       example2.chpl
+
+.. note::
+
+  Users can set environment variables like ``LDFLAGS`` for ``-L`` arguments and
+  ``CFLAGS`` for ``-I`` arguments, to avoid throwing these flags every time.
+
+  Additionally, the required linker flags (``-l``) may vary depending on the
+  ``LAPACK`` implementation being used.
+
+**Building programs with optional dependencies**
+
+.. code-block:: chpl
+
+  // example3.chpl
+  var A = Matrix(3,5);
+  A = 2;
+  var AA = A.dot(A.T);
+
+
+The program above uses :proc:`dot`, which has two available implementations:
+one that depends on  :mod:`BLAS` and one that is written in Chapel.  The
+program will default to using the more performant ``BLAS`` implementation
+of matrix-matrix multiplication.
+
+Following the instructions from the
+:mod:`BLAS` module documentation, the above program could be compiled if
+``BLAS`` is available on the system and specified with the following
+compilation flags:
 
 .. code-block:: bash
 
-  chpl --set blasImpl=none --set lapackImpl=none myProgram.chpl
+  # Building with BLAS dependency
+  chpl -I$PATH_TO_CBLAS_DIR \
+       -L$PATH_TO_BLAS_LIBS -lblas \
+       example3.chpl
+
+.. note::
```

Comment:
I think this is a duplicate note

Reply:
It's the same note as before, except for `BLAS` instead of `LAPACK`. Maybe I should just add a single note towards the end covering both of them?

Reply:
I would merge them into 1 note